tdrs-backend/tdpservice/users/api/login.py

```diff
@@ -207,3 +251,102 @@ def get(self, request, *args, **kwargs):
+
+        decoded_id_token = self.decode_jwt(id_token, settings.LOGIN_GOV_ISSUER, settings.LOGIN_GOV_CLIENT_ID, cert_str,
+                                           options)
+        return {"id_token": decoded_id_token}
+
+    def get_token_endpoint_response(self, code):
+        """Build out the query string params and full URL path for token endpoint."""
+        try:
+            options = {
+                "client_assertion": generate_client_assertion(),
+                "client_assertion_type": settings.LOGIN_GOV_CLIENT_ASSERTION_TYPE
+            }
+            token_params = generate_token_endpoint_parameters(code, options)
+            token_endpoint = settings.LOGIN_GOV_TOKEN_ENDPOINT + "?" + token_params
+            return requests.post(token_endpoint)
+
+        except ValueError as e:
+            logger.exception(e)
+            return Response(
+                {
+                    "error": str(e)
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+    def get_auth_options(self, access_token, sub):
+        """Add specific auth properties for the CustomAuthentication handler."""
+        auth_options = {"login_gov_uuid": sub}
+        return auth_options
+
+
+class TokenAuthorizationAMS(TokenAuthorizationOIDC):
+    """Define methods for handling login request from HHS AMS."""
+
+    def decode_payload(self, token_data, options=None):
+        """Decode the payload with keys for AMS."""
+        id_token = token_data.get("id_token")
+        access_token = token_data.get("access_token")
+
+        ams_configuration = LoginRedirectAMS.get_ams_configuration()
+        certs_endpoint = ams_configuration["jwks_uri"]
+        cert_str = generate_jwt_from_jwks(certs_endpoint)
+        issuer = ams_configuration["issuer"]
+        audience = settings.AMS_CLIENT_ID
+
+        decoded_id_token = self.decode_jwt(id_token, issuer, audience, cert_str, {"verify_aud": False})
+        decoded_access_token = self.decode_jwt(access_token, issuer, audience, cert_str, {"verify_aud": False})
+
+        return {
+            "id_token": decoded_id_token,
+            "access_token": access_token,
+            "decoded_access_token": decoded_access_token
+        }
+
+    def get_token_endpoint_response(self, code):
+        """Build out the query string params and full URL path for token endpoint."""
+        # First fetch the token endpoint from AMS.
+        ams_configuration = LoginRedirectAMS.get_ams_configuration()
+        options = {
+            "client_id": settings.AMS_CLIENT_ID,
+            "client_secret": settings.AMS_CLIENT_SECRET,
+            "scope": "openid+email+profile",
+            "code": code,
+            "grant_type": "authorization_code",
+            "redirect_uri": settings.BASE_URL + "/oidc/ams",
+        }
+
+        token_endpoint = ams_configuration["token_endpoint"]
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
```

Comment:
Comparing this function to the loginDotGov one, should we either explain why headers are necessary only here or also add this there too? Not sure if it's worth it to you to deploy new code just for testing login.gov flows with new headers when it is already functional.